### ISSUE TITLE
Learning and transferring mid-level image representations using convolutional neural networks

### ISSUE THREAD

**Issue by sooooner** [sooooner/papers#12]
# Learning and transferring mid-level image representations using convolutional neural networks[[link](https://www.cv-foundation.org/openaccess/content_cvpr_2014/papers/Oquab_Learning_and_Transferring_2014_CVPR_paper.pdf)]

## Introduction
유사성이 없는 데이터로 훈련된 모델으로 transfer learning 함
imagenet으로 훈련하고 Pascal VOC에 적용

## Transferring CNN weights
CNN은 매우 많은 매개 변수를 가짐, 수천개의 데이터에 매우 많은 매개 변수를 직접 학습하는것은 문제가 될수 있음  
핵심 아이디어는 CNN 내부 layer가 middle level image represention의 추출기 역할을 할 수 있다는 것.
**The key idea of this work is that the internal layers of the CNN can act as a generic extractor of mid-level image representation, which can be pre-trained on one dataset  and then re-used on other target tasks**

하지만 이는 서로 다른 labels 와 이미지의 분포 때문에 어렵다. 때문에 다음과 같은 작업을 한다.
1. source와 target task 사이의 클래스 레이블을 명시적으로 다시 매핑하는 아키텍처 설계
2. source와 target task에서 객체 크기, 위치, 장면의 다양한 분포를 명시적으로 처리하는 절차(슬라이딩 윈도우에서 영감을 받음)

### Network architecture
model : AlexNet
source : imagenet
target : Pascal VOC

target task의 경우 대상 class에 대한 score를 출력하거나 이미지가 class에 없는경우 background를 출력한느 네트워크 설계할거.
그러나 source task의 label은 target task와 매우 다를수 있음('label bias') 
예를 들어 source network는 허스키, 진돗개등 다양한 개의 label이 존재하지만 target task는 단순히 '개' 라고 되있을 수 있음
이는 action classification에서 더욱 분명해진다고 함

이를 위해 source network의 마지막 FC8을 제거하고 target network에 FCa, FCb를 추가함 여기서 C1~5, FC6~7은 고정하고 학습함

![image](https://user-images.githubusercontent.com/31437443/107847044-b983be00-6e2b-11eb-879f-c66e5787b656.png)

### Network training
두 데이터의 분포 사이즈등이 다름('a dataset capture bias'), source에 없는 obj가 target에는 있을수 있음
-> 슬라이딩 윈도우 활용

각 이미지에서 S by S size 의 500여개 patch 추출, (S = min(w, h)/λ),  λ ∈ {1,1.3,1.6,2,2.4,2.8,3.2,3.6,4} 
이후 inputs으로 넣기 위해 224*224로 rescale함(각 patch는 50% 이상 overlap이 존재함)

샘플링된 이미지 패치는 하나이상의 obj, background 또는 obj의 일부만 포함될 수 있음. train image에서 patch에 label을 지정하기 위해 patch P의 bounding box와 이미지에서 주석이 달린 obj의 Ground Truth Bounding box B사이의 overlap을 측정

patch는 class o 에 해당하는 Bo가 있는 경우 클래스 o에 대한 positive training example로 지정됨
여기서 class o는
1. |P ∩ Bo| ≥ 0.2|P|
2. |P ∩ Bo| ≥ 0.6|Bo|
3. 겹치지 않음

위의 정의에서 | A | bounding box A의 면적을 측정합니다. 라벨링 기준은 아래 그림 에 나와 있습니다.

![image](https://user-images.githubusercontent.com/31437443/107847800-010d4880-6e32-11eb-98b3-e2c7238cb6bb.png)

#### Dealing with background
위에서 설명한것 처럼 target task에는 obj를 포함하지 않는 patch에 대한 추가 background label이 존재함
또 하나의 어려운 점은 training data가 불균형 하다는 것. 
training data의 대부분 patch는  background에서 나옴 이는 훈련 비용 함수의 가중치를 다시 지정함으로써 해결할 수 있음
이는 훈련중 그레디언트 가중치를 다시 지정하는 것과 같음 
약간 다른 절차를 선택하고 대신 훈련 패치를 다시 샘플링 하여 훈련 데이터 분포의 균형을 맞출거임
리샘플링된 traning set는 SGD를 위한 mini batch를 형성하는데 사용됨 이는 training background patch의 10% random sampling으로 구현됨

## Classification
다음 공식을 사용하여 이미지의 obj Cn에 대한 전체 점수를 계산
![image](https://user-images.githubusercontent.com/31437443/107848018-c2788d80-6e33-11eb-9660-be773c087bd9.png)

y(Cn|Pi) is the output of the network for class Cn on image patch Pi, M is the number of patches in the image, and k ≥ 1 is a parameter. Higher values of k focus on the highest scoring patches and attenuate the contributions

## Experiment
imagenet으로 훈련시킨 AlexNet을 PASCAL VOC에 적용
성능 향상이 극적이진 않지만 모델이 robust한 결과를 얻음

## Conclusion
간단한 transfer learning이 더 작은 크기의 어려운 벤치마크 데이터 세트에서 좋은 성능을 얻었다. 
imagenet에서 훈련한 CNN에서 추출한 midle-level-feature의 높은 잠재력 입증
